client.go

```diff
@@ -9,13 +9,65 @@ import (
 	"mime"
 	"mime/multipart"
 	"net/http"
+	"sync"
+	"time"
 )
 
 // Multipart object returned by AVS.
 type responsePart struct {
 	Directive *Message
 }
 
+// blockingReader enables streaming request.Audio without ending HTTP request prematurely
+type blockingReader struct {
+	buf *bytes.Buffer
+	mut *sync.Mutex
+	eof bool
+	err error
+}
```

Comment:
This is a pretty complex type for piping data. See my comment below in the `Do` function about a possible alternative.